source/time_tables/previous_passages.cpp

```diff
@@ -71,7 +70,7 @@ previous_passages(const std::string &request,
 
     for(auto dt_stop_time : passages_dt_st) {
         pbnavitia::Passage * passage;
-        if(stop_event == StopEvent::pick_up) {
```

Comment:
This change was made in prevision of a re-factor and merge with next_passages.
The second reason for this is semantic : pick_up is far less ambiguous than departure (as one can think of trip departure in that case).
So some re-factor is still missing, but this was the direction I took.